models/staging/stg_nhl__live_plays.sql

```diff
@@ -212,7 +212,79 @@ live_plays as (
         , bp.event_id
         , bp.player_id
         , bp.team_id
-
+        -- last play metrics (fixed on 8/20/2023)
+        , case
+            when lag(bp.event_idx, 1) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 1) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 2) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 2) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 3) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 3) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 4) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 4) over (partition by game_id order by bp.game_id, event_idx, play_time)
+            when lag(bp.event_idx, 5) over (partition by game_id order by bp.game_id, event_idx, player_index, play_time) != bp.event_idx
+                then lag(bp.event_idx, 5) over (partition by game_id order by bp.game_id, event_idx, play_time)
+        end as last_play_event_idx
```

Comment:
Can you help me follow what's happening here?

Reply:
oh, yea sure although it's incredibly confusing :)

so our original raw play-by-play data stored each play as an array... when we unlisted such that each player's involvement in the play had its own row, it unlisted in order of involvement... so like, goalscorer = row 1, primary assist = row 2, secondary assist row =3, etc... this logic here is basically to associate the last play with a player (see pic)